### DISCUSSION TITLE
How to configure drivebase for gyro

### DISCUSSION THREAD

**Discussion by sanenerdBmg** [pybricks/support#1894] [General]
I am using latest beta:
I am brief because I am teaching some kids a class on this and have to attend to them

('essentialhub', '3.6.0b2', 'ci-release-81-v3.6.0b2 on 2024-10-15')

I am experiencing that this code spins my bot forever:
Also seen weird issue on spike prime where motor.track_target(90) does nothing.

```python
from pybricks.hubs import EssentialHub
from pybricks.parameters import Axis, Direction, Port
from pybricks.pupdevices import Motor
from pybricks.robotics import DriveBase

# Set up all devices.
essential_hub = EssentialHub(top_side=Axis.Z, front_side=-Axis.X)
motor = Motor(Port.A, Direction.COUNTERCLOCKWISE)
motor_2 = Motor(Port.B, Direction.CLOCKWISE)
drive_base = DriveBase(motor, motor_2, 56, 128)


# The main program starts here.
drive_base.use_gyro(True)
drive_base.reset()
drive_base.turn(90)
```

**Comment by laurensvalk** [comment]
> I am experiencing that this code spins my bot forever:

Carefully check which motor is on the left and which is on the right.

If the motor you chose as left is configured to be on the right, it would spin the opposite way as you are seeing.

> motor.track_target(90) does nothing.

When the program ends, the motors stop. So if this is the last thing in your program, that makes it seem like nothing happens. Try adding a wait after it to make sure your program did not end.

**Reply by sanenerdBmg** [reply]
Thanks, I wanted to send the smallest possible code that illustrates the issue

**Reply by sanenerdBmg** [reply]
Yes, I was an idiot - the bot has gears that reverses the clockwise/counterclockwise setting on the hub - thanks for your help

**Reply by laurensvalk** [reply]
No worries! That's how we learn :smile: 

This is a good example of what a "positive feedback" loop does (bad), as opposed to a "negative feedback" (good).

When configured the opposite way, the robot will try to correct a clockwise deviation by going even further clockwise (and vice versa), making it worse and worse, and so on, so it spins in circles.
